backend/src/main/java/gov/cdc/usds/simplereport/service/dataloader/PatientLinkDataLoader.java

```diff
@@ -0,0 +1,53 @@
+package gov.cdc.usds.simplereport.service.dataloader;
+
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.db.repository.PatientLinkRepository;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PatientLinkDataLoader extends KeyedDataLoaderFactory<UUID, PatientLink> {
+  public static final String KEY = "testResults[*].patientLink";
+
+  @Override
+  public String getKey() {
+    return KEY;
+  }
+
+  public static PatientLink getMostRecentPatientLink(List<PatientLink> patientLinks) {
+    if (patientLinks == null) {
+      return null;
+    }
+    patientLinks.sort(
+        (pl1, pl2) -> {
+          if (pl1.getCreatedAt().equals(pl2.getCreatedAt())) {
+            return 0;
+          } else if (pl1.getCreatedAt().before(pl2.getCreatedAt())) {
+            return 1;
+          } else return -1;
+        });
+    return patientLinks.get(0);
```

Comment:
Since we're only using the most recent, would there be further performance benefit to just finding the most recent time in one pass instead of sorting?  Guess that depends on the size of the lists?